### ISSUE TITLE
NullCheckStrategy.Always with Generics producing non compiling code

### ISSUE THREAD

**Issue by gabron** [mapstruct/mapstruct#1680]
Version: 1.3.0.Beta2
Java 10

Using the NullCheckStrategy.Always produces non compiling code in the null check code block in the generated source. 

I noticed that when using @MapppingTarget in a method on an interface that has a @Mapper(nullValueCheckStrategy=NullValueCheckStrategy.ALWAYS), and the class being mapped contains a generic with the wild card that extends a class, I get an error in the generated class pertaining to that generic field.

For example, the following class:
```
public class Artist {
	private List<? extends String> genres; //getters-setters
}
```

When used with the following mapper interface:
```
@Mapper(
		componentModel="spring", 
		nullValueCheckStrategy=NullValueCheckStrategy.ALWAYS, 
		nullValuePropertyMappingStrategy=NullValuePropertyMappingStrategy.IGNORE)
public interface IArtistUpdateMapper {
	Artist updateArtistFromArtistDto(ArtistDto source, @MappingTarget Artist target);
}
```

Provides the following generated code:
```
if ( target.getGenres() != null ) {
            List<String> list1 = source.getGenres();
            if ( list1 != null ) {
                target.getGenres().clear();
                target.getGenres().addAll( list1 );
            }
        }
        else {
            List<String> list1 = source.getGenres();
            if ( list1 != null ) {
                target.setGenres( new ArrayList<String>( list1 ) );
        }
}
```

The addAll method produces the following compiler error in the generated source:
`The method addAll(Collection<? extends capture#12-of ? extends String>) in the type List<capture#12-of ? extends String> is not applicable for the arguments (List<String>)`

If I remove the wildcard `private List<String> genres;` it will work just fine

**Comment by chris922** [comment]
Is `? extends String` now just an example and you've got the same error with a different class as well? 

... just asking, because `? extends String` doesn't make any sense, as `String` is final.

**Comment by gabron** [comment]
Yeah, that was just an example, I was getting it with a different class as
well but didn't want to include it for simplicity sake.

**Comment by sjaakd** [comment]
Is this a duplicate of #542 ??
